### PR TITLE
Update Glossary Terms

### PR DESCRIPTION
**Pull request by shardhrv**
## Description

This PR updates the glossary as required.

## Changes
- Updated docs/DeveloperGuide.md to include newly included terms in user guide

## Issue
#40 

## Additional Notes
Let me know if any further changes are needed.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -330,6 +330,18 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
 * **Mainstream OS**: Windows, Linux, Unix, MacOS
 * **Private contact detail**: A contact detail that is not meant to be shared with others
+* **PlanPal**: A software system designed to manage contact information specifically for student event planners, supporting functions like adding, updating, and deleting contacts, with data stored locally.
+* **CLI (Command Line Interface)**: A text-based user interface where the user interacts with the system by typing commands.
+* **Contact**: An individual or entity whose information is stored in PlanPal, typically including name, phone number, email, address, role, and optionally a Telegram handle.
+* **Role**: A category assigned to a contact to indicate their relationship to an event, such as "attendee," "vendor," "volunteer," or "sponsor."
```
Comment (brianchoon) [inline docs/DeveloperGuide.md:336]: Nit: the full stop should be outside the double quotes

**Review by brianchoon — COMMENTED** [review]
Great work! I have left a comment for a single nit.

**Comment by jan-kai1** [comment]
LGTM

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (shardhrv) [reply docs/DeveloperGuide.md]: Nice catch, thanks!

**Review by brianchoon — APPROVED** [review]
LGTM!

**Pull request merged**